### DISCUSSION TITLE
grab automatically generated filename from csvwriter so it can be used for later appending?

### DISCUSSION THREAD

**Discussion by preinagel** [bonsai-rx/bonsai#1584] [Q&A] [answered]
Hi, 
I want to append the trial parameters for all the trials of a session to a single csv file, but create a new file for each new session with an automatically generated timestamp suffix.

My idea was to create the file at the top of the workflow by sending some string to csvwriter setting that csvwriter node suffix setting to "timestamp"; and then use createPropertySource from that node to publish the automatically generated filename for later use, like this:

![image](https://github.com/bonsai-rx/bonsai/assets/80356291/9bcb4366-42df-481f-9430-fda46d96d32f)

(such that I could use that subject as an input to an externalized filename property of other csvwriter nodes set to Append=True).

But no string never shows up in the FileName node. Is there some way to grab that string?  

Or some better way to achieve the original goal?

**Comment by bruno-f-cruz** [comment, marked as answer]
https://github.com/orgs/bonsai-rx/discussions/1308#discussioncomment-5443949

This might work?

**Reply by preinagel** [reply]
Yep, or at least that led me to a solution, which looks like this:
![image](https://github.com/bonsai-rx/bonsai/assets/80356291/a082ced5-a2f4-4398-9fbc-621e1a1acfa5)

where the timer is just for demo purposes, standing in for whatever data I want to log.

If anyone else wants to use it or see the details:
[SessionFilename.zip](https://github.com/bonsai-rx/bonsai/files/12875320/SessionFilename.zip)
